### PR TITLE
Fixes Content Type builder when the autoReload feature is disabled

### LINKED ISSUE #5038: Content type creation fails silently in production mode
When Strapi is started with `strapi start` you can't create new content types in the admin panel, apparently (`strapi dev` works, however). When you try to create a content type in production mode anyway, Strapi will just fail silently on save.

**Expected behaviour**: An error message should pop up, saying that new content types cannot be created in production mode and `strapi dev` should be used instead. Or even better: Gray out everything that's not usable and display a notification.

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
This PR fixes the content type builder plugin when the autoReload feature is disabled.
The plugin is still accessible but it will be like in production mode (with no call to action displayed) but a notification will pop up to tell the user to start his server with `strapi develop`.

#### My PR is a:

- [ ] 💥 Breaking change
- [x] 🐛 Bug fix Fixes #5038 
- [ ] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [x] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: packages/strapi-plugin-content-type-builder/admin/src/translations/en.json [hunk 1]
```diff
@@ -132,6 +132,7 @@
   "modelPage.attribute.relationWith": "Relation with",
   "modelPage.contentHeader.emptyDescription.description": "There is no description",
   "notification.info.creating.notSaved": "Please save your work before creating a new content type or component",
+  "notification.info.autoreaload-disable": "The autoReload feature is required to use this plugin. Start your server with strapi develop",
```
Comment (derrickmehaffy) [inline packages/strapi-plugin-content-type-builder/admin/src/translations/en.json]: can we do escaped code blog here? like 
```
\`strapi develop\`
```

**Comment by Aurelsicoko** [comment]
LGTM

File: packages/strapi-admin/admin/src/containers/Admin/index.js [hunk 2]
```diff
@@ -144,6 +144,7 @@ export class Admin extends React.Component {
 
     return (
       <GlobalContextProvider
+        autoReload={this.props.global.autoReload}
```
Comment (HichamELBSI) [inline packages/strapi-admin/admin/src/containers/Admin/index.js]: `this.props.global` is already destructured in the render method. You should destructure `this.props.global.autoReload`  too.

File: packages/strapi-plugin-content-type-builder/admin/src/containers/DataManagerProvider/index.js [hunk 3]
```diff
@@ -59,7 +60,10 @@ const DataManagerProvider = ({ allIcons, children }) => {
   const componentMatch = useRouteMatch(
     `/plugins/${pluginId}/component-categories/:categoryUid/:componentUid`
   );
-  const isInDevelopmentMode = currentEnvironment === 'development';
+  const formatMessageRef = useRef();
```
Comment (HichamELBSI) [inline packages/strapi-plugin-content-type-builder/admin/src/containers/DataManagerProvider/index.js]: Is the `useRef()` really it necessary here?

Comment (soupette) [reply packages/strapi-plugin-content-type-builder/admin/src/containers/DataManagerProvider/index.js]: Yes so this way we don't have to pass it in the dependencies array of the `useEffect` hook so it avoids the stale closure issue.

File: packages/strapi-admin/admin/src/containers/Admin/index.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-admin/admin/src/containers/Admin/index.js]: You are right

**Comment by HichamELBSI** [comment]
LGTM

**Pull request merged**